walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,136 @@
+        Map<String, Boolean> attendeesMap = new HashMap<String, Boolean>();
+        for (String attendee : attendees) { 
+            attendeesMap.put(attendee, true); 
+        }
+        for (String attendee : optionalAttendees) { 
+            attendeesMap.put(attendee, false); 
+        }
+
+        System.out.println(attendeesMap.toString());
+
+        /* Events stored in Map to allow to dynamically remove events in loop (so as not to 
+           work on the same event that is shared by multiple attendees) */
+        Map<Event, Boolean> eventsMap = new HashMap<Event, Boolean>();
+        for (Event event : events) { 
+            eventsMap.put(event, true); 
+        }
+
+        Iterator eventsIterator; 
+        for (Map.Entry<String, Boolean> attendeeEntry : attendeesMap.entrySet()) {
+            String attendee = attendeeEntry.getKey();
+
+            eventsIterator = eventsMap.entrySet().iterator();
+            while(eventsIterator.hasNext()) {
+
+                Map.Entry pair = (Map.Entry) eventsIterator.next();
+                if ((Boolean) pair.getValue()) {
+                    Event event = (Event) pair.getKey();
+
+                    Collection<String> eventAttendees = event.getAttendees();
+                    if (eventAttendees.contains(attendee)) {
+                        TimeRange timeRange = event.getWhen();
+                        blockTimeRange(timeRange, request.getDuration(), attendeeEntry.getValue());
+
+                        if (attendeeEntry.getValue()) {
+                            eventsIterator.remove();
+                        }      
+                        // If there are only optional attendees & no times are available for them
+                        if (!attendeeEntry.getValue() && availableTimes.isEmpty()) {
+                            availableTimes.add(TimeRange.WHOLE_DAY);
+                        }
+                    }
+                } 
+            }
+        } 
+        return availableTimes;
+    }
+
+    public void blockTimeRange(TimeRange blockedTimeRange, long meetingDuration, boolean mandatory) {
+        for (int i = 0; i < availableTimes.size(); i++) {
+            TimeRange timeRange = availableTimes.get(i);
+
+            if ((blockedTimeRange.equals(TimeRange.WHOLE_DAY) && !mandatory) 
+                || blockedTimeRange.duration() < meetingDuration) return;
+
+            if (timeRange.overlaps(blockedTimeRange)) {
+
+                TimeRange newTimeRange1 = TimeRange.fromStartEnd(timeRange.start(), 
+                                                blockedTimeRange.start(), false);
+                TimeRange newTimeRange2 = TimeRange.fromStartEnd(blockedTimeRange.end(), 
+                                                timeRange.end(), false);      
+                /*
+                            |--blocked--|
+                        |-----timeRange----|
+                */
+                if (timeRange.contains(blockedTimeRange)) {
```

Comment:
What's the difference with this `contains` check and the `overlaps` check above?